pyanaconda/modules/storage/partitioning/automatic/automatic_partitioning.py

```diff
@@ -56,6 +64,117 @@ def _get_initialization_config(self):
+        devices = []
+        for root in storage.roots:
+            if mountpoint in root.mounts:
+                devices.append(root.mounts[mountpoint])
+        if len(devices) > 1:
+            raise StorageError(_("Multiple devices found for mountpoint '{}': {}")
+                               .format(mountpoint,
+                                       ", ".join([device.name for device in devices])))
+        if not devices:
+            if required:
+                raise StorageError(_("No devices found for mountpoint '{}'").format(mountpoint))
+            else:
+                return None
+
+        return devices[0]
+
+    def _reused_devices_mountpoints(self, request):
+        return request.reused_mount_points + request.erased_mount_points
+
+    def _get_reused_device_names(self, storage):
+        reused_devices = {}
+        for mountpoint in self._reused_devices_mountpoints(self._request):
+            device = self._get_mountpoint_device(storage, mountpoint)
+            reused_devices[device.name] = mountpoint
+        return reused_devices
+
+    def _erase_mountpoint(self, storage, mountpoint):
+        device = self._get_mountpoint_device(storage, mountpoint)
+        log.debug("DDDDD erase device %s for  mountpoint: %s", device, mountpoint)
+        reused_devices = self._get_reused_device_names(storage)
+        # FIXME it would be better to really erase the device
+        reformat_device(storage, device, dependencies=reused_devices)
+
+    def _remove_mountpoint(self, storage, mountpoint):
+        device = self._get_mountpoint_device(storage, mountpoint, required=False)
+        if device:
+            log.debug("DDDDD remove device %s for mountpoint %s", device, mountpoint)
+            destroy_device(storage, device)
+        else:
+            log.debug("DDDDD device to be removed for mountpoint %s not found", mountpoint)
+
+    def _remove_bootloader_partitions(self, storage, required=True):
+        log.debug("DDDDD platform.partitions %s", platform.partitions)
+
+        bootloader_types = ["efi", "biosboot", "appleboot", "prepboot"]
+        bootloader_parts = [part for part in platform.partitions if part.fstype in bootloader_types]
+        if len(bootloader_parts) > 1:
+            raise StorageError(_("Multiple bootloader partitions required: %s"), bootloader_parts)
+        if not bootloader_parts:
+            log.debug("No bootloader partition required")
+            return False
+        part_type = bootloader_parts[0].fstype
+
+        devices = []
+        for device in storage.devices:
+            if device.format.type == part_type:
+                devices.append(device)
+        if len(devices) > 1:
+            raise StorageError(_("Multiple devices found for bootloader partition '{}': {}")
+                               .format(part_type,
+                                       ", ".join([device.name for device in devices])))
```

Comment:
So does this fail with two disks both with a BIOS boot partition? Because that's not a problem (it's not necessary, but also doesn't break anything).

Reply:
Good point, I'll think about loosing the requirement.

Reply:
At this point I'd be rather more restrictive (->safe?). I'd think finding 2 bios boot partitions won't be very common case and I'd rather handle it if/when we hit it. Also 2 bios boot partitions suggests to me there might be more complex setup so rather we should be careful (and force the user to deselect one of the disks for example).
But I didn't think this through enough yet.